src/browser/setupWorker/start/createStartHandler.ts

```diff
@@ -118,35 +118,42 @@ If this message still persists after updating, please report an issue: https://g
         return registration
       }
 
-    registerWorker().then((pendingWorkerRegistration) => {
-      // If the activation was cancelled (e.g. by calling "worker.stop()"),
-      // short-circuit: nothing to return.
-      if (pendingWorkerRegistration == null) {
-        return
-      }
+    registerWorker()
+      .then(async (pendingWorkerRegistration) => {
+        // If the activation was cancelled (e.g. by calling "worker.stop()"),
+        // short-circuit: nothing to return.
+        if (pendingWorkerRegistration == null) {
+          return
+        }
 
-      const pendingWorker =
-        pendingWorkerRegistration.installing ||
-        pendingWorkerRegistration.waiting
-
-      // Wait until the worker is activated.
-      // Assume the worker is already activated if there's no pending registration
-      // (i.e. when reloading the page after a successful activation).
-      if (pendingWorker) {
-        const stateChangeListener = () => {
-          if (pendingWorker.state === 'activated') {
-            workerActivationPromise.resolve(pendingWorkerRegistration)
-
-            pendingWorker.removeEventListener(
-              'statechange',
-              stateChangeListener,
-            )
-          }
+        const pendingWorker =
+          pendingWorkerRegistration.installing ||
+          pendingWorkerRegistration.waiting
+
+        // Wait until the worker is activated.
+        // Assume the worker is already activated if there's no pending registration
+        // (i.e. when reloading the page after a successful activation).
+        if (pendingWorker) {
+          await new Promise<void>((resolve) => {
+            const stateChangeListener = () => {
+              if (pendingWorker.state === 'activated') {
+                pendingWorker.removeEventListener(
+                  'statechange',
+                  stateChangeListener,
+                )
+                resolve()
+              }
+            }
+
+            pendingWorker.addEventListener('statechange', stateChangeListener)
+          })
         }
 
-        pendingWorker.addEventListener('statechange', stateChangeListener)
-      }
-    })
+        workerActivationPromise.resolve(pendingWorkerRegistration)
+      })
+      .catch((error) => {
+        workerActivationPromise.reject(error)
```

Comment:
when registering the worker throws, we also need to reject the worker activation, I think?